src/vs/editor/contrib/format/formatActions.ts

```diff
@@ -381,7 +381,12 @@ export class FormatDocumentAction extends EditorAction {
 		const { tabSize, insertSpaces } = editor.getModel().getOptions();
 		return formatDocument(workerService, editor, { tabSize, insertSpaces }, CancellationToken.None).catch(err => {
 			if (NoProviderError.is(err)) {
-				notificationService.info(nls.localize('no.documentprovider', "There is no document formatter for '{0}'-files installed.", editor.getModel().getLanguageIdentifier().language));
+				const language = editor.getModel().getLanguageIdentifier().language;
+				if (language === 'plaintext') {
+					notificationService.info(nls.localize('no.documentprovider', "There is no document formatter for '{0}'-files installed.", language));
+				} else {
+					notificationService.info(nls.localize('no.documentprovider.suggestlink', "There is no document formatter for '{0}'-files installed. [Search for '{0}' formatter in Marketplace](https://marketplace.visualstudio.com/search?term={0}&target=VSCode&category=Formatters&sortBy=Relevance)", language));
+				}
```

Comment:
You cannot hardcode that in the OOS repo. That's why we have a `product.json`-file. Then, I don't think it's best to open a browser given that we have a viewlet for extensions. @sandy081 might know how to get that pre-populated with formatters for a language

Reply:
I mean I know the string needed to populate `@category:formatters $LANGUAGE`. Maybe even used already call to extensions viewlet with population. But no idea how to do that from Notification link. Should the button be used for that?

Reply:
Maybe a command link works, otherwise a button isn't a bad choice for this.

Reply:
Can I just wait until someone implements the said command?

Reply:
You can also add it as part of this PR :)